pkg/client/src/app/pages/applications/analysis-wizard/components/add-custom-rules.tsx

```diff
@@ -71,37 +77,73 @@ export const AddCustomRules: React.FunctionComponent<IAddCustomRules> = ({
     setReadFileData(newReadFiles);
   };
 
-  // callback that will be called by the react dropzone with the newly dropped file objects
+  const isSchemaValid = (value: string) => {
+    const validationResult = xmllint.xmllint.validateXML({
+      xml: value,
+      schema: XSDSchema,
+    });
+
+    if (!validationResult.errors) return true;
+    return false;
+  };
+
+  const validateXMLFile = (data: string) => {
+    // Filter out "data:text/xml;base64," from data
+    const payload = atob(data.substring(21));
+
+    let isXML = false;
+    let isXSD = false;
+
+    const validationObject = XMLValidator.validate(payload, {
+      allowBooleanAttributes: true,
+    });
+
+    if (validationObject === true) {
+      isXML = true;
```

Comment:
Should we use react state for these values?

Reply:
There is no need for state as the validation process is not transient across rendering.